server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java

```diff
@@ -170,4 +170,67 @@ public void testInitial() throws RetryAttemptException {
         assertFalse(details.get(1).isInitial());
     }
 
+    @Test
+    public void testCalculateActualTimeSync() throws RetryAttemptException {
```

Comment:
`testCalculateActualTimeAsync` I believe satisfies the second and third cases? the second child duration (700) is greater than the parent's duration (500), and consequently it ends after the parent node (your third case).

The test `testCalculateActualTimeSync` has two children, one after the other - I don't mind adding a test for the first case, but wondering if there was a special reason?